### PR TITLE
[AMORO-3038] Unify the API set for dashboard and open API

### LINKED ISSUE #3038: [Improvement]: Unify the API set for dashboard and open API
### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/amoro/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

Following discussion:https://lists.apache.org/thread/h1sxgz26mndoq077wlbj8sk2w0zxn1pb

### How should we improve?

Unify the API set for both dashboard and Open API

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Subtasks

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by mansonliwh**
## Why are the changes needed?

Close #3038.

## Brief change log

Unify the API set for both dashboard and Open API in the same url [ /api/ams/v1 ]. 

Previously, frontend requests and external interface requests were distinguished by using the paths **ams/v1** and **api/ams/v1** respectively. Frontend requests required cookie validation for login status, while external interface requests needed to pass an apikey parameter for verification. In this design, the request paths from both sources are unified to **api/ams/v1**. For frontend requests, an additional header parameter **_X-Request-Source: frontend_** will be included. This allows the source of the request to be distinguished and routed to different validation logic accordingly.

## How was this patch tested?

- [ ☑️ ] Add screenshots for manual tests if appropriate
image

- [ ☑️ ] Run test locally before making a pull request

### CODE REVIEW

**Review by czy006 — COMMENTED** [review]
Thank you for your contribution. Do we have any relevant documents. should we introduce Swagger and related SDK in other PR?

File: amoro-ams/src/main/java/org/apache/amoro/server/dashboard/DashboardServer.java [hunk 1]
```diff
@@ -355,7 +354,8 @@ private EndpointGroup apiGroup() {
 
   public void preHandleRequest(Context ctx) {
     String uriPath = ctx.path();
-    if (needApiKeyCheck(uriPath)) {
+    String requestSource = ctx.header("X-Request-Source");
+    if (needApiKeyCheck(uriPath) && !"frontend".equals(requestSource)) {
```
Comment (zhoujinsong) [inline amoro-ams/src/main/java/org/apache/amoro/server/dashboard/DashboardServer.java:358]: Is `Web` a proper name for requests from the front end?

And we may better declare a static final variable for it, and the same to `X-Request-Source`

**Review by zhoujinsong — COMMENTED** [review]
Thanks a lot for your work.
I left a small suggestion, PTAL.

**Review by zhoujinsong — APPROVED** [review]
LGTM.

Thanks a lot for pushing forward this improvement!

**Pull request merged**